core/src/test/scala/org/http4s/parser/QueryParserSpec.scala

```diff
@@ -25,7 +25,7 @@ class QueryParserSpec extends Http4sSpec {
     }
 
     "return an empty Map for an empty query string" in {
-      parseQueryString("") must be_\/-(Query.empty)
+      parseQueryString("") must be_\/-(Query.none)
```

Comment:
Well, I renamed `EmptyQuery` to `NoQuery` and `Query.empty` to `Query.none`, but cases like this one makes me think that simple change alone isn't a perfect fit.  May have to rethink what an empty query vs a 0-length one.  If I create a FormQuery based on `Map()` or `Vector()` should that yield a 0-length query or a `NoQuery`?  Feels like a `NoQuery` to me.

Should `parseQueryString("")` produce a 0-length query, or a NoQuery?  Feels like a 0-length query to me, which `.asForm` should be Map("" -> Nil).  Hopefully it holds up.